hack/functests.sh

```diff
@@ -38,14 +38,45 @@ fi
 rm -rf $ARTIFACTS
 mkdir -p $ARTIFACTS
 
-if [[ ${KUBEVIRT_PROVIDER} =~ .*(k8s-1\.16)|(k8s-1\.17)|k8s-sriov.* ]]; then
-    echo "Will skip test asserting the cluster is in dual-stack mode."
-    FUNC_TEST_ARGS="${FUNC_TEST_ARGS} -skip-dual-stack-test"
-fi
+function functest() {
+    if [[ ${KUBEVIRT_PROVIDER} =~ .*(k8s-1\.16)|(k8s-1\.17)|k8s-sriov.* ]]; then
+        echo "Will skip test asserting the cluster is in dual-stack mode."
+        extra_args="-skip-dual-stack-test"
+    fi
+    _out/tests/ginkgo -r $@ _out/tests/tests.test -- ${extra_args} -kubeconfig=${kubeconfig} -container-tag=${docker_tag} -container-tag-alt=${docker_tag_alt} -container-prefix=${functest_docker_prefix} -image-prefix-alt=${image_prefix_alt} -oc-path=${oc} -kubectl-path=${kubectl} -gocli-path=${gocli} -installed-namespace=${namespace} -previous-release-tag=${PREVIOUS_RELEASE_TAG} -previous-release-registry=${previous_release_registry} -deploy-testing-infra=${deploy_testing_infra} -config=${KUBEVIRT_DIR}/tests/default-config.json --artifacts=${ARTIFACTS}
```

Comment:
> > As can be seen in hack/functest.sh, first the parallel ones are done (the test suite cleans up itself, as usual), then the serial ones are started. Here I slightly diverge from my previous statement that I want all tests executed. Since we can now already execute in this early state 250 tests in parallel (pretty much half of the tests), I am fine with failing early after the serial tests. We will see how this goes, but I am at least slightly optimistic that for development and flakefinder CI will still provide enough output to be valuable. I am not 100% sure that this is really the best option. May be changed again later.
> 
> I think we did not have to deal with post test run cleanup leftovers until now. Or did we?

Not in CI on merged code. Only in the sense that we would see issues a lot on `make cluster-sync` or when running `make functest` multiple times in a row, which I do a lot locally (and others probably too). And so far the CI runs did not show anything like that.

> Just wandering if we may fall into cases where the previous parallel run left some leftovers and did not wait for them to cleanup before continuing.

The test suite creates almost not resources outside of the test namespaces. If it succeeded, it should work pretty much 100% at the moment. Nothing indicates anything different so far. That can of course change with every PR.

> 
> > At the end, if something in the CI environment causes issues, there is nothing wrong with the introduced changes.
> 
> The introduced changes just bring challenges to the debugging and troubleshooting steps. It also introduces new awareness need from contributors and reviewers on how to deal with it.
> e2e tests are flaky because they require interaction between components, which are time/delay sensitive. When adding concurrency to the matrix, things get one level more complex and potentially flakier.

Yes I took only simple tests so far. All CI runs so far did not indicate any additional flakyness (also not less). What I meant above is, even if we have to disable the parallel test runs due to CI issues, there is nothing as such wrong with this PR.

Running the tests in parallel was always a goal and k8s and others do it successfully too. There is no reason to believe that the complexity regarding to debugging for our tests is so much higher (at least not in a non-fixable way). Creating more clusters in parallel is another angle on this, we can/should leverage that too for some cases, but this is one important step in general. One can run without almost any additional resource consumption run half of the kubevirt e2e tests in less than 30 minutes. Spinning up a second cluster has huge overhead compared to that. I would still consider it, for instance to run tests which need to run in serial in another cluster in parallel.